x/retry/retry_test.go

```diff
@@ -781,6 +782,7 @@ func TestMiddleware(t *testing.T) {
 		t.Run(tt.msg, func(t *testing.T) {
 			retry := NewUnaryMiddleware(
 				WithPolicyProvider(tt.policyProvider),
+				TallyScope(tally.NoopScope),
```

Comment:
This can probably be done in a different diff but you'll want to use a tally TestScope to verify the correct metrics were emitted in the different cases above.